### PR TITLE
ddns-scripts: Add dnspod.cn

### PR DESCRIPTION
**Pull request by zhujunsan**
Maintainer: San / @zhujunsan
Compile tested: should be universal, FriendlyElec NanoPi R2S, OpenWrt 21.02
Run tested: aarch64_generic?, FriendlyElec NanoPi R2S, OpenWrt 21.02, tests done

Description:
ddns-scripts: add support for dnspod.cn api

> Thanks to https://github.com/nixonli/ddns-scripts_dnspod and maybe more people before that

### CODE REVIEW

**Comment by neheb** [comment]
ping @feckert

File: net/ddns-scripts/files/usr/lib/ddns/update_dnspod_cn.sh [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+	sleep 3
+	add_domain
+elif [ $ret = 2 ];then
+	sleep 3
+	update_domain
+else
+	printf "%s\n" " $(date +%H%M%S)       : Record needs not update: [Record IP:$__RECIP] [Local IP:$__IP]" >> $LOGFILE
+fi
+
+return 0
```
Comment (feckert) [inline net/ddns-scripts/files/usr/lib/ddns/update_dnspod_cn.sh:137]: New line missing on end of file

File: net/ddns-scripts/files/usr/share/ddns/default/dnspod.cn.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
+	"name": "dnspod.cn",
+	"ipv4": {
+        	"url": "update_dnspod_cn.sh"
+	},
+	"ipv6": {
+        	"url": "update_dnspod_cn.sh"
+	}
+}
```
Comment (feckert) [inline net/ddns-scripts/files/usr/share/ddns/default/dnspod.cn.json:9]: New line missing on end of file

File: net/ddns-scripts/files/usr/lib/ddns/update_dnspod_cn.sh [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+# Check inputs
+[ -z "$username" ] && write_log 14 "Configuration error! [User name] cannot be empty"
+[ -z "$password" ] && write_log 14 "Configuration error! [Password] cannot be empty"
+
+# Check external tools
+[ -n "$CURL_SSL" ] || write_log 13 "Dnspod communication require cURL with SSL support. Please install"
+[ -n "$CURL_PROXY" ] || write_log 13 "cURL: libcurl compiled without Proxy support"
+
+# Declare variables
+local __URLBASE __HOST __DOMAIN __TYPE __CMDBASE __POST __POST1 __RECIP __RECID __TTL
```
Comment (feckert) [inline net/ddns-scripts/files/usr/lib/ddns/update_dnspod_cn.sh:12]: As far as I know ash does not allow a `local` definition without a function

**Review by feckert — CHANGES_REQUESTED** [review]
Please also bump the PKG_RELEASE number

File: net/ddns-scripts/files/usr/lib/ddns/update_dnspod_cn.sh (hunk 3 above)
Comment (zhujunsan) [reply net/ddns-scripts/files/usr/lib/ddns/update_dnspod_cn.sh]: As not really needed, I just comment out the line.
(See similar usage in other scripts in the same folder, maybe need a bigger rework)

**Review by feckert — CHANGES_REQUESTED** [review]
Please bump the PKG_RELEASE

**Comment by BKPepe** [comment]
You should squash your changes into the single one and use your real name in the `Signed-off-by`.

**Comment by zhujunsan** [comment]
> You should squash your changes into the single one and use your real name in the `Signed-off-by`.

Done

**Comment by zhujunsan** [comment]
rebased to most up to date master branch code. php error has gone, musl compile error still exist. Don't think that has anything to do with this commit thought  : (

**Pull request merged**